### PR TITLE
SC-19093 support custom logs from agent

### PR DESCRIPTION
**Pull request by cpipilas**
Add a new page for setting up custom log shipping directly from a configuration file in Sematext Agent

NOTE: merge after 3.6.0 release

### CODE REVIEW

File: docs/agents/sematext-agent/custom-logs.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+## Requirements
+
+Sematext Agent [version 3.6.0](https://sematext.com/docs/agents/sematext-agent/releasenotes/#version-360) and newer.
+
+## How to start
+
+[Create a new Logs App](https://sematext.com/docs/logs/quick-start/) in Sematext Cloud. If you have custom log files, create a [Generic Logs App](https://sematext.com/docs/integration/generic-logs-integration/). For log files generated from one of our [supported integrations](https://sematext.com/docs/integration/), create a Logs App of the relevant type.
+
+As soon as the Sematext Agent is installed, ignore the Discovery screen and keep a copy of the App token. To retrieve this token, navigate to the **Ship Logs** screen from the left Menu panel and select **Custom Integrations**.
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:18]: Screenshot would be handy here.

File: docs/agents/sematext-agent/custom-logs.md [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+
+The next step is to create the `custom-logs.yml` file and place it in the appropriate directory (see below) in the agent configuration.
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:20]: STA will probably come with a sample custom-logs.yml, but with everything commented out. If so, then create --> edit.

File: docs/agents/sematext-agent/custom-logs.md [hunk 3]
```diff
@@ -0,0 +1,179 @@
+title: Sematext Agent Custom Logs Configuration
+description: Custom Logs Configuration for Sematext Agent
+
+Sematext Agent supports configuring log shipping through a YAML file, `custom-logs.yml`. This feature allows users to specify log file locations and custom parsing rules, which is useful for log files not [discovered automatically](https://sematext.com/docs/fleet/discovery/) or when custom log parsing rules are needed.
+
+## How it works
+
+The Sematext Agent periodically checks the `custom-logs.yml` file. If valid log shipping rules are found, the agent starts collecting, parsing, and shipping logs from the specified files to the Sematext Cloud Logs App that matches the provided App token. To stop log shipping, remove the rule from the file.
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:8]: Probably worth explicitly saying one can configure N log files, N parsing rules, N destination Logs Apps.

File: docs/agents/sematext-agent/custom-logs.md [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+
+## Configuration Details
+
+Each section in the `custom-logs.yml` file defines a log shipping rule. There are also several variables to configure how the Sematext Agent collects and parses your log files.
+
+### Log type
+
+- **Description**: The type of your logs. You can define your own type or use a pre-defined type.
+- **Required**: Yes
+- **Example**: `myapp-logs` which is a custom type, `nginx` which is pre-defined
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:50]: custom log type name needs to be all alphanumerics, plus dash, or ....?

File: docs/agents/sematext-agent/custom-logs.md [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+| Varnish Cache                        | varnishcache             |
+
+### `files`
+
+- **Description**: Specifies the paths to the log files. Glob patterns are not supported.
+- **Required**: Yes
+- **Example**:
+
+```yaml
+  files: ["/path/to/myapp.log"]
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:76]: Show example with more than 1 file?

File: docs/agents/sematext-agent/custom-logs.md [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+- **Required**: Yes
+- **Example**:
+
+```yaml
+  token: "af526914-15fe-481e-b06c-e7e07daf793c"
+```
+
+### `isJSON`
+
+- **Description**: Set to `true` if your log files are JSON formatted. If `true`, `regexPatterns` and `patternTypes` are not needed.
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:91]: are not needed.... and if specified will be ignored

File: docs/agents/sematext-agent/custom-logs.md [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+- **Required**: No
+- **Default**: `false`
+- **Example**:
+
+  ```yaml
+  isJSON: true
+  ```
+
+### `regexPatterns`
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:100]: So we want to say what happens if one specifies a valid regex?

File: docs/agents/sematext-agent/custom-logs.md [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+
+Using a pre-defined type for Nginx logs:
+
+```yaml
+nginx:
+  files: ["/var/log/nginx/access.log"]
+  token: "3d542016-4255-4de4-9b69-03b6066e0155"
+```
+
+### JSON Formatted Logs
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:136]: Do we want to say that JSON needs to be on one line and not pretty printed?

File: docs/agents/sematext-agent/custom-logs.md [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+```yaml
+a-json-logs-app:
+  files: ["/var/log/myapp/myapp.json", "/var/log/myapp/errors.json"]
+  isJSON: true
+  token: "af526914-15fe-481e-b06c-e7e07daf793c"
+```
+
+### Custom Log Format
+
+Let's say we have two log files named `myapp.log` and `errors.log` in the `/var/log/myapp/`` folder that contain log events like the ones below:
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:149]: Extra back tick

File: docs/agents/sematext-agent/custom-logs.md [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+- **Description**: Defines the regex patterns to parse logs. If defined, pre-defined types won't be used. Regular expressions are wrapped with single quotes (r'...'). The value between the quotes uses the [Rust regex syntax](https://docs.rs/regex/latest/regex/#syntax).
+- **Required**: No
+- **Example**:
+
+```yaml
+  regexPatterns:
+    - '^(?P<timestamp>\d{4}-\d{2}-\d{2}[\s|T][\d+|:]+.\d+)\s(?P<message>[\s|\S]+)'
+```
+
+### `patternTypes`
```
Comment (otisg) [inline docs/agents/sematext-agent/custom-logs.md:111]: Should we list all valid pattern types?

**Review by otisg — CHANGES_REQUESTED** [review]
Looks good. N smaller suggestions made.

File: docs/agents/sematext-agent/custom-logs.md (hunk 10 above)
Comment (boratanrikulu) [reply docs/agents/sematext-agent/custom-logs.md:111]: TODO(bora): check valid  patterns.

File: docs/agents/sematext-agent/custom-logs.md [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+| Elasticsearch                        | elasticsearch            |
+| OpenSearch                           | opensearch               |
+| JVM apps                             | jvm                      |
+| Linux daemons                        | linux-daemon             |
+| MySQL                                | mysql                    |
+| Nginx                                | nginx                    |
+| Postgres                             | postgresql               |
+| RabbitMQ                             | rabbitmq                 |
+| Solr                                 | solr                     |
+| Varnish Cache                        | varnishcache             |
```
Comment (boratanrikulu) [inline docs/agents/sematext-agent/custom-logs.md:67]: TODO(bora): add this page's link to the yaml file.

File: docs/agents/sematext-agent/custom-logs.md (hunk 2 above)
Comment (boratanrikulu) [reply docs/agents/sematext-agent/custom-logs.md:20]: Yes they will have a sample file.
TODO(bora): 
1. 
``` 
  cp ./configs/st-agent.yml $IMAGE_PATH/opt/spm/properties/st-agent.yml
  cp ./configs/autodisco.yml $IMAGE_PATH/opt/spm/properties/autodisco.yml
...
  getGithubResource "st-agent.yml" "$IMAGE_PATH/opt/spm/properties/st-agent.yml"
  getGithubResource "autodisco.yml" "$IMAGE_PATH/opt/spm/properties/autodisco.yml"
```
2.
```
cp $SPM_HOME/properties/st-agent.yml $DIAG_DIR/st-agent 2> /dev/null
cp $SPM_HOME/properties/autodisco.yml $DIAG_DIR/st-agent 2> /dev/null
```

Comment (boratanrikulu) [reply docs/agents/sematext-agent/custom-logs.md:20]: TODO(bora): spm files and diag

File: docs/agents/sematext-agent/custom-logs.md (hunk 10 above)
Comment (cpipilas) [reply docs/agents/sematext-agent/custom-logs.md:121]: Aren't we covered by [Rust regex syntax](https://docs.rs/regex/latest/regex/#syntax) mentioned above?

File: docs/agents/sematext-agent/custom-logs.md (hunk 2 above)
Comment (cpipilas) [reply docs/agents/sematext-agent/custom-logs.md]: Updated. Keeping this unresolved until Bora is ready.

File: docs/agents/sematext-agent/custom-logs.md (hunk 10 above)
Comment (otisg) [reply docs/agents/sematext-agent/custom-logs.md:121]: I was referring to "string", "int"... what are the other ones?

File: docs/agents/sematext-agent/custom-logs.md (hunk 2 above)
Comment (boratanrikulu) [reply docs/agents/sematext-agent/custom-logs.md]: https://github.com/sematext/spm-client/pull/147

File: docs/agents/sematext-agent/custom-logs.md (hunk 7 above)
Comment (cpipilas) [reply docs/agents/sematext-agent/custom-logs.md:110]: maybe https://regex101.com? cc @boratanrikulu

File: docs/agents/sematext-agent/custom-logs.md (hunk 10 above)
Comment (boratanrikulu) [reply docs/agents/sematext-agent/custom-logs.md:121]: "string", "int", and "float"

Comment (boratanrikulu) [reply docs/agents/sematext-agent/custom-logs.md:121]: and "bool"

File: docs/agents/sematext-agent/custom-logs.md [hunk 12]
```diff
@@ -0,0 +1,189 @@
...
+myapp-logs:
+  files:
+    - "/var/log/myapp/myapp.log"
+    - "/var/log/myapp/errors.log"
+  token: "0d63b278-4eae-40de-8862-db6b490634db"
+  regexPatterns:
+    - '^(?P<timestamp>\d{4}/\d{2}/\d{2}\s[\d|:]+)\s\[(?P<level>.+?)]\s(?P<pid>\d+)#(?P<tid>\d+):\s(?P<message>.*)'
+  patternTypes:
+    timestamp: "string"
+    severity: "string"
```
Comment (boratanrikulu) [inline docs/agents/sematext-agent/custom-logs.md:180]:
```suggestion
    level: "string"
```

File: docs/agents/sematext-agent/custom-logs.md (hunk 7 above)
Comment (boratanrikulu) [reply docs/agents/sematext-agent/custom-logs.md:110]: could be this one https://rustexp.lpil.uk/

File: docs/agents/sematext-agent/custom-logs.md [hunk 13]
```diff
@@ -0,0 +1,189 @@
...
+    level: "string"
+    pid: "int"
+    tid: "int"
+    message: "string"
+```
+
+## Troubleshooting tips and known issues
+
+- Support for Sematext Agent for Windows is coming soon
+- Support for Kubernetes and containerized environments is coming soon
```
Comment (boratanrikulu) [inline docs/agents/sematext-agent/custom-logs.md:189]:
```suggestion
- Support for Windows is coming soon
- Support for Kubernetes and containerized environments is coming soon
```

**Review by boratanrikulu — APPROVED** [review]
LGTM

**Pull request merged**